challengeA/ishiwata/lib/plan.rb

```diff
@@ -0,0 +1,62 @@
+class Plan
+
+  HALFPLAN = ["従量電灯B", "ずっとも電気1"]
+  MINPLAN = ["従量電灯B"]
+  MINCHARGE = [235]
+
+  attr_reader :plans, :demand_charge, :energy_charge
+
+  def initialize(args)
+    @plans = args[:plans] || defaults
+    @demand_charge = DemandCharge.new(args[:ampere])
+    @energy_charge = EnergyCharge.new(args[:usage])
+  end
+
+
+  def show_plan
+    plans.map{ |plan| plan.merge( sum_charge(plan)) }
+  end
+
+  def sum_charge(plan)
+    demand = demand_charge.choose_plan(plan[:plan_name])
+    energy = energy_charge.choose_plan(plan[:plan_name])
+    if demand.nil? || energy.nil?
```

Comment:
エラー処理は guard 節にする。

https://techracho.bpsinc.jp/hachi8833/2016_10_11/26950

```suggestion
    return notice if demand.nil? || energy.nil?
```

Reply:
この場合に関して言えば、専用のエラーをraise するようにした上で
上位で、rescue するように設計すれば、
`{ charge: 計算結果 }` というハッシュではなく、直接
計算結果を返すようにできる。
その方がより直接的で読みやすいインタフェース。